components/elm/src/main/elm_varpar.F90

```diff
@@ -48,20 +48,23 @@ module elm_varpar
   integer, parameter :: ndst        =   4     ! number of dust size classes (BGC only)
   integer, parameter :: dst_src_nbr =   3     ! number of size distns in src soil (BGC only)
   integer, parameter :: sz_nbr      = 200     ! number of sub-grid bins in large bin of dust size distribution (BGC only)
-  integer, parameter :: mxpft       =  50     ! maximum number of PFT's for any mode;
+  !integer, parameter :: mxpft       =  50     ! maximum number of PFT's for any mode;
+  integer :: mxpft                  =  50     ! maximum number of PFT's for any mode;
+                                              ! can be modified from reading pft-physiology in 'main/pftvarcon.F90:pftconrd'
   ! FIX(RF,032414) might we set some of these automatically from reading pft-physiology?
   integer, parameter :: numveg      =  16     ! number of veg types (without specific crop)
   integer, parameter :: nlayer      =   3     ! number of VIC soil layer --Added by AWang
   integer            :: nlayert               ! number of VIC soil layer + 3 lower thermal layers
 
-  integer :: numpft      = mxpft   ! actual # of patches (without bare), a total that spans LUs
+  integer :: numpft      =  50     ! actual # of patches (without bare), a total that spans LUs
```

Comment:
Might be safer to initialize this to a nonsense number (a negative perhaps?), so that it crashes if there is some weird scenario where it does not get set, is it interpreted from reading in real data, right?

Reply:
mxpft is set by reading in real data. numpft is still used as an upper limit. We are working out how to simplify this but I'd like to do it in a later PR - it is a long-standing tangle of usage that this PR makes a little better, but doesn't go all the way to cleaning up.